odoo/addons/base/models/res_users.py

```diff
@@ -981,11 +984,12 @@ def _assert_can_auth(self):
             _logger.warning(
                 "Login attempt ignored for %s on %s: "
                 "%d failures since last success, last failure at %s. "
+                "User: %s. "
                 "You can configure the number of login failures before a "
                 "user is put on cooldown as well as the duration in the "
                 "System Parameters. Disable this feature by setting "
                 "\"base.login_cooldown_after\" to 0.",
-                source, self.env.cr.dbname, failures, previous)
+                source, self.env.cr.dbname, failures, previous, uid or login or "Unknown")
```

Comment:
Maybe use a fallback which can't be a login for clarity e.g. "?"? Otherwise things might get weird if there's a db with an actual "Unknown" user.

Also `%r` might be a good placeholder, especially to avoid confusion due to the closing period.

Reply:
And is having two different parameters really useful? Just have a single `user` which can be either uid or login, it's not like that matters hugely.

Finally not sure that's looking great since it ends up as a single line, this is going to log as

> Login attempt ignored for 123.456.789.123 on test. 2 failures since last success, last failure at 12:34. User: 5. You can configure the number of login failures...

From a readability and a convenience standpoint I think it'd make sense to add the new information to the first sentence which is most of the useful content, maybe something like

    ignored for %s by %r

or

   ignored for %s (%s)

Though there's the drawback that an existing regex might break.

Reply:
Thanks for the review!